scripts/proteinInteractionEBI/parseEBI.py

```diff
@@ -0,0 +1,331 @@
+    schemaPieceList.append(curLine)
+    
+    return '\n'.join(schemaPieceList), termDic['id'][0], dcid, newSourceMap
+
+
+def main(argv):
+    dbFile = argv[0]
+    with open(dbFile, 'r') as fp:
+        file = fp.read()
+    # clip exists in dcs already. Substitute with ClipInteraction
+    file = file.replace('name: clip\ndef', 'name: clip interaction\ndef')
+    fileTerms = file.split('\n\n')[1:]
+
+    '''
+    Parsing Steps:
+    1. build the tree by the psi-mi number. A dictionary {psi-mi: node} is used 
+    to access nodes as well.
+    2. save all the tree nodes in the subtree of the three nodes into three set:
+        id: MI:0001 name: interaction detection method
+        id: MI:0190 name: interaction type
+        id: MI:0444 name: database citation
+    3. save the nodes in the three sets to the corresponding enumearation schema
+    '''
+    
+    id2node = {}
+    id2className = {}
+    # build nodes and create the id2node dictionary at first iteration
+    for termText in fileTerms:
+        if not termText.startswith('[Term]'):
+            continue
+        # idString example: "MI:0000"
+        idString = termText.split('\n')[1].split(' ')[1]
+        className = get_class_name(termText.split('\n')[2].split(': ')[1])
+        id2className[idString] = className
+        id2node[idString] = Node(idString)
+
+    # build the parent-child relation at the second iteration
+    for termText in fileTerms:
+        if not termText.startswith('[Term]'):
+            continue
+        termList = termText.split('\n')
+        idString = termList[1].split(' ')[1]
+        parentIdList = get_parent_id_list(termList[1:])
+        for pId in parentIdList:
+            id2node[pId].childList.append(id2node[idString])
+            id2node[idString].parentList.append(id2node[pId])
+
+    # get the idStrings for the three target set
+    dfsCaller = GetTreeValues()
+    interactionTypeIdSet = dfsCaller.get_subset_id(id2node['MI:0001']) # root id: MI:0001 
+    detectionMethodIdSet = dfsCaller.get_subset_id(id2node['MI:0190'])# root id: MI:0190
+    interactionSourceIdset = dfsCaller.get_subset_id(id2node['MI:0444']) # root id: MI:0444
+
+    # delete root node value from the set
+    interactionTypeIdSet.remove('MI:0001')
+    detectionMethodIdSet.remove('MI:0190')
+    interactionSourceIdset.remove('MI:0444')
+
+    setList = [interactionTypeIdSet, detectionMethodIdSet,interactionSourceIdset]
+    print('The schema amount of each Enum: interactionType, detectionMethod,interactionSource are ')
+    print([len(s) for s in setList])
+
+    schemaList = []
+    psimi2dcid = []
+    newSourceMap = {'references':{}}
+    for termText in fileTerms:
+        if not termText.startswith('[Term]'):
+            continue
+        term = termText.split('\n')[1:]
+        schemaRes = get_schema_from_text(term,id2node,newSourceMap,id2className,
+                                      interactionTypeIdSet,detectionMethodIdSet,interactionSourceIdset)
```

Comment:
term, id2node, newSourceMap

If you can use pylint - http://google.github.io/styleguide/pyguide.html